### PR TITLE
[ENG-2306] Four new registration forms

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Adds new schemas for OSF preregistration

## Changes

- 4 new schemas and a migrations

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify all language is the same as as the excel spreadsheet
- Verify all schemas are initially invisible

## Documentation

simple tech task, no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2306

### CODE REVIEW

File: website/project/metadata/real-world-evidence.json [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+               "required":true
+            }
+         ]
+      },
+      {
+         "id":"page5",
+         "title":"Protocol Document",
+         "questions":[
+            {
+               "qid":"q20",
```
Comment (UdayVarkhedkar) [inline website/project/metadata/real-world-evidence.json:223]: This look like it should be required

**Review by UdayVarkhedkar — CHANGES_REQUESTED** [review]
Just needs a few small corrections to bring it in line with the spreadsheets defining the schemas.

**Review by UdayVarkhedkar — APPROVED** [review]

File: website/project/metadata/asist-hypothesis-capability-registration.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+   "name":"ASIST Hypothesis/Capability Registration",
+   "version":1,
+   "config":{
+      "hasFiles":true
+   },
+   "description":"This template was created by researchers from the qualitative community for preregistration of qualitative research",
```
Comment (brianjgeiger) [inline website/project/metadata/asist-hypothesis-capability-registration.json:7]: According to [the ticket](https://openscience.atlassian.net/browse/ENG-2306), this should be, "This preregistration template is part of the DARPA Artificial Social Intelligence for Successful Teams (ASIST) program. This template is intended to formally register performer hypothesis and AI capability preregistrations throughout the ASIST program."

File: website/project/metadata/asist-hypothesis-capability-registration.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      {
+         "id":"page1",
+         "title":"Research Aims",
+         "questions":[
+            {
+               "qid":"q1",
+               "type":"string",
+               "format":"text",
+               "title":"Study registration link",
+               "description":"Please specify the overall purposes, objectives or aims of the research.\n\nIn addition, please reflect on whether your aim is different across different domains (e.g. knowledge generation, policy development, community resourcing). If so, specify your aim for each domain that is relevant for your study.",
```
Comment (brianjgeiger) [inline website/project/metadata/asist-hypothesis-capability-registration.json:18]: Where did this text come from? The spreadsheet I have looks more like this:

Screen Shot 2020-10-22 at 4 03 05 PM

File: website/project/metadata/asist-hypothesis-capability-registration.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+               "required":true
+            },
+            {
+               "qid":"q2",
+               "title":"Hypothesis/Capability Registration",
+               "type":"osf-upload",
+               "format":"osf-upload-open",
+               "description":"Please upload your pre-analysis plan, along with any other supporting documents, such as survey instrument, research protocol, any data, etc. OSF functionality allows for de-identified registration links to be generated after the registration is complete. However, it cannot automatically update the documents you attach.<b>If you plan to generate a de-identified link for this registration (for journal submission/review purposes, for instance), please MAKE SURE ALL FILES UPLOADED HERE ARE ANONYMIZED</b>",
```
Comment (brianjgeiger) [inline website/project/metadata/asist-hypothesis-capability-registration.json:26]: This also does not match with what I see in the spreadsheet.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
The information I have is wildly different from what I'm seeing here. Am I missing something? If I am, could you point me to the correct information? If I'm not, can you go back through and double-check everything to make sure it's all good?

Also, one of these registrations should not be invisible.

File: website/project/metadata/real-world-evidence.json [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+               "format":"textarea",
+               "title":"Other",
+               "required":false
+            },
+            {
+               "qid":"q8",
+               "type":"string",
+               "format":"textarea",
+               "title":"Study population",
+               "description":"Please choose one the population under evaluation in the study",
```
Comment (brianjgeiger) [inline website/project/metadata/real-world-evidence.json:92]: This was wrong in the spreadsheet. It should be, "The population under evaluation in the study"

File: website/project/metadata/real-world-evidence.json [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+               "help":" The cohort entry date was defined by new initiation of SLEEPINATOR – XR (intervention) or melatonin (comparator).",
+               "required":true
+            },
+            {
+               "qid":"q10",
+               "type":"string",
+               "format":"textarea",
+               "title":"Specific Inclusion criteria",
+               "description":"A description of how exactly patients enter the study",
+               "help":"a prior diagnosis of condition X.",
```
Comment (brianjgeiger) [inline website/project/metadata/real-world-evidence.json:111]: Capitalize "A"

File: website/project/metadata/real-world-evidence.json [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+               "required":true
+            },
+            {
+               "qid":"q11",
+               "type":"string",
+               "format":"textarea",
+               "title":"Specific Inclusion criteria",
```
Comment (brianjgeiger) [inline website/project/metadata/real-world-evidence.json:118]: "Specific Exclusion criteria"

File: website/project/metadata/real-world-evidence.json [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+               "options":[
+                  "No data handling at this point",
+                  "Acquired access to the source data",
+                  "Conducted feasibility counts",
+                  "Created an analytic dataset for the study population of interest",
+                  "Conducted preliminary analyses, including descriptive statistics and observation of data distributions – but no analyses of association between treatment and outcome",
+                  "Study analysis in progress",
+                  "Statistical comparisons completed"
+               ],
+               "description":"TPlease choose the one best short description that best reflects the amount of data handling that you have had up until the date of registration for this study.",
```
Comment (brianjgeiger) [inline website/project/metadata/real-world-evidence.json:212]: Typo: "Please" not "TPlease"

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
I did a more thorough pass. Here's everything I could find.

**Comment by brianjgeiger** [comment]
Based on the [last travis run](https://travis-ci.org/github/CenterForOpenScience/osf.io/builds/738353570) it looks like this might not be up-to-date with develop and its migrations.

File: osf/migrations/0221_add_schemas.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+    schema_names = [schema['name'] for schema in schemas]
+
+    RegistrationSchema.objects.filter(name__in=schema_names).update(visible=False, active=False)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('osf', '0218_auto_20200929_1850'),
```
Comment (brianjgeiger) [inline osf/migrations/0221_add_schemas.py:27]: 0219_auto_20201020_1836.py

Comment (brianjgeiger) [inline osf/migrations/0221_add_schemas.py:27]:
```suggestion
        ('osf', '0219_auto_20201020_1836'),
```

**Review by brianjgeiger — APPROVED** [review]

**Pull request merged**